File: library/src/main/java/me/yamlee/jsbridge/jscall/SendParamsProcessor.java
package me.yamlee.jsbridge.jscall;

import org.json.JSONException;
import org.json.JSONObject;

import me.yamlee.jsbridge.BaseJsCallProcessor;
import me.yamlee.jsbridge.JsCallData;
import me.yamlee.jsbridge.NativeComponentProvider;
import me.yamlee.jsbridge.WVJBWebViewClient;
import timber.log.Timber;

/**
 * Js获取原生传过来的参数
 *
 * @author yamlee
 */
public class SendParamsProcessor extends BaseJsCallProcessor {
    public static final String FUNC_NAME = "getParams";

    public SendParamsProcessor(NativeComponentProvider componentProvider) {
        super(componentProvider);
    }

    @Override
    public boolean onHandleJsQuest(JsCallData callData) {
        if (FUNC_NAME.equals(callData.getFunc())) {
            String params = callData.getParams();
            try {
                JSONObject jsonObject = new JSONObject(params);
                String module = jsonObject.optString("module");
                Timber.i("module is" + module);
            } catch (JSONException e) {
                e.printStackTrace();
            }

            return true;
        }
        return false;
    }

    @Override
    public boolean onResponse(WVJBWebViewClient.WVJBResponseCallback callback) {
        callback.callback(new String(""));
        return true;
    }

    @Override
    public String getFuncName() {
        return FUNC_NAME;
    }
}

File: android-jsBridgeCore/src/main/java/me/yamlee/jsbridge/jscall/SendParamsProcessor.kt
package me.yamlee.jsbridge.jscall

import me.yamlee.jsbridge.BaseJsCallProcessor
import me.yamlee.jsbridge.JsCallData
import me.yamlee.jsbridge.JsCallback
import me.yamlee.jsbridge.NativeComponentProvider
import me.yamlee.jsbridge.utils.LogUtil
import org.json.JSONException
import org.json.JSONObject

/**
 * Js获取原生传过来的参数
 *
 * @author yamlee
 */
class SendParamsProcessor(componentProvider: NativeComponentProvider) : BaseJsCallProcessor(componentProvider) {

    override fun onHandleJsQuest(callData: JsCallData, callback: JsCallback): Boolean {
        if (FUNC_NAME == callData.func) {
            val params = callData.params
            try {
                val jsonObject = JSONObject(params)
                val module = jsonObject.optString("module")
                LogUtil.info("module is$module")
            } catch (e: JSONException) {
                e.printStackTrace()
            }

            return true
        }
        return false
    }

    override fun getFuncName(): String {
        return FUNC_NAME
    }

    companion object {
        const val FUNC_NAME = "getParams"
    }
}
